src/snowflake/connector/cache.py

```diff
@@ -219,7 +236,7 @@ def _update(
                 g = iter(other.items())
             to_insert = {k: CacheEntry(expiry=expiry, entry=v) for k, v in g}
         elif isinstance(other, SFDictCache):
-            other._clear_expired_entries()
+            other.clear_expired_entries()
```

Comment:
here, `self._lock` is locked (line 227), then why are we not using `other._clear_expired_entries()` here? `clear_expired_entries` tries to acquire lock again

Reply:
nvm, other has a different lock